src/Content.php

```diff
@@ -238,7 +262,7 @@ public function setValues(array $values)
         );
         // Check if the values need to be unserialized, and pre-processed.
         foreach ($this->values as $key => $value) {
-            if (in_array($this->fieldtype($key), $serializedFieldTypes)) {
+            if ((in_array($this->fieldtype($key), $serializedFieldTypes)) || ($key == 'templatefields')) {
```

Comment:
Isn't 'templatefields' is a serializedFieldTypes value… add it to the array above, no?